Error while checking build/O2/fullCI for 484aff8091d90b5e9b1d6da59b9feb3f18aa30c1 at 2023-11-03 13:11:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile digi.log
digi.log:[ERROR] Workflow crashed - pid 27432 (GRPUpdater) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27359 (ITSDigitizer) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27382 (TPCDigitizer_29) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27374 (TPCDigitizer_21) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27386 (TPCDigitizer_31) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27365 (TPCDigitizer_12) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27379 (TPCDigitizer_26) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27402 (TPCDigitizer_8) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27405 (TPCDigitizer_9) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27352 (TPCDigitizer_0) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27398 (TPCDigitizer_7) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27371 (TPCDigitizer_18) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27373 (TPCDigitizer_20) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27387 (TPCDigitizer_4) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27394 (TPCDigitizer_6) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27381 (TPCDigitizer_28) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
digi.log:[ERROR] Workflow crashed - pid 27363 (TPCDigitizer_10) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27432 (GRPUpdater) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27359 (ITSDigitizer) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27382 (TPCDigitizer_29) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27374 (TPCDigitizer_21) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27386 (TPCDigitizer_31) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27365 (TPCDigitizer_12) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27379 (TPCDigitizer_26) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27402 (TPCDigitizer_8) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[ERROR] Workflow crashed - pid 27405 (TPCDigitizer_9) was killed abnormally with exit code 128, could be out of memory killer, segfault, unhandled exception, SIGKILL, etc...
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12200/484aff8091d90b5e9b1d6da59b9feb3f18aa30c1/build_O2_fullCI/pretty.html).

--- a/Framework/Core/include/Framework/DeviceContext.h
+++ b/Framework/Core/include/Framework/DeviceContext.h
@@ -16,6 +16,7 @@
 #define O2_FRAMEWORK_DEVICECONTEXT_H_
 
 typedef struct uv_timer_s uv_timer_t;
+typedef struct uv_signal_s uv_signal_t;
 
 namespace o2::framework
 {
@@ -27,6 +28,7 @@ struct ComputingQuotaStats;
 struct DeviceContext {
   ComputingQuotaStats* quotaStats = nullptr;
   uv_timer_t* gracePeriodTimer = nullptr;
+  uv_signal_t* sigusr1Handle = nullptr;
   int expectedRegionCallbacks = 0;
   int exitTransitionTimeout = 0;
 };

--- a/Framework/Core/src/DataProcessingDevice.cxx
+++ b/Framework/Core/src/DataProcessingDevice.cxx
@@ -493,6 +493,10 @@ void on_signal_callback(uv_signal_t* handle, int signum)
   ZoneScopedN("Signal callaback");
   LOG(debug) << "Signal " << signum << " received.";
   auto* registry = (ServiceRegistry*)handle->data;
+  if (!registry) {
+    LOG(debug) << "No registry active. Ignoring signal";
+    return;
+  }
   ServiceRegistryRef ref{*registry};
   auto& state = ref.get<DeviceState>();
   auto& quotaEvaluator = ref.get<ComputingQuotaEvaluator>();
@@ -972,10 +976,14 @@ void DataProcessingDevice::InitTask()
   // an event from the outside, making sure that the event loop can
   // be unblocked (e.g. by a quitting DPL driver) even when there
   // is no data pending to be processed.
-  auto* sigusr1Handle = (uv_signal_t*)malloc(sizeof(uv_signal_t));
-  uv_signal_init(state.loop, sigusr1Handle);
-  sigusr1Handle->data = &mServiceRegistry;
-  uv_signal_start(sigusr1Handle, on_signal_callback, SIGUSR1);
+  if (deviceContext.sigusr1Handle == nullptr) {
+    deviceContext.sigusr1Handle = (uv_signal_t*)malloc(sizeof(uv_signal_t));
+    deviceContext.sigusr1Handle->data = &mServiceRegistry;
+    uv_signal_init(state.loop, deviceContext.sigusr1Handle);
+    uv_signal_start(deviceContext.sigusr1Handle, on_signal_callback, SIGUSR1);
+  }
+  // When we start, we must make sure that we do listen to the signal
+  deviceContext.sigusr1Handle->data = &mServiceRegistry;
 
   /// Initialise the pollers
   DataProcessingDevice::initPollers();
@@ -1675,6 +1683,14 @@ void DataProcessingDevice::ResetTask()
 {
   ServiceRegistryRef ref{mServiceRegistry};
   ref.get<DataRelayer>().clear();
+  auto& deviceContext = ref.get<DeviceContext>();
+  // If the signal handler is there, we should
+  // hide the registry from it, so that we do not
+  // end up calling the signal handler on something
+  // which might not be there anymore.
+  if (deviceContext.sigusr1Handle) {
+    deviceContext.sigusr1Handle->data = nullptr;
+  }
 }
 
 struct WaitBackpressurePolicy {

